Include typescript in the project

yon-cuadrado opened the pull request.

christian-bromann commented on tsconfig.json:
```diff
@@ -0,0 +1,33 @@
+        "declaration": true,
+        "inlineSources": true,
+        "typeRoots": ["./node_modules/@types","./lib/types" ],
+        "rootDir": "./",
```
I don't thing we need any files within the root directory no?
```suggestion
        "rootDir": "./lib",
```

christian-bromann commented on package.json:
```diff
@@ -2,17 +2,21 @@
   "version": "2.0.4",
   "description": "A WDIO reporter that creates CucumberJS JSON files for WebdriverIO V5+",
-  "main": "./build/reporter.js",
+  "main": "./dist/lib/reporter.js",
```
so we can change this to 
```suggestion
  "main": "./dist/reporter.js",
```

christian-bromann commented on lib/types/wdio.d.ts:
```diff
@@ -0,0 +1,79 @@
+import { WriteStream } from 'fs';
+import { cjson_metadata } from '../models';
+
+declare module '@wdio/types' {
```
Can we just extend the types in the WebdriverIO project rather then redeclaring them here?

yon-cuadrado replied:
I have substituted all the modified types with extended ones

christian-bromann reviewed (commented):
Excellent work !

Added a couple of comments.

christian-bromann reviewed (approved):
This is awesome, thank you so much 👍

christian-bromann commented:
> [ ](/yon-cuadrado)
> ` [added logfile](/webdriverio-community/wdio-cucumberjs-json-reporter/pull/56/commits/eb94fc35b8409c6ab12bcf264db1a00b329a04ce)`

Can we revert this change or is this file necessary?

yon-cuadrado commented:
> > [ ](/yon-cuadrado)
> > ` [added logfile](/webdriverio-community/wdio-cucumberjs-json-reporter/pull/56/commits/eb94fc35b8409c6ab12bcf264db1a00b329a04ce)`
> 
> Can we revert this change or is this file necessary?
The tests pass in local without this file. I have added it to see if the error of these tests: https://github.com/webdriverio-community/wdio-cucumberjs-json-reporter/runs/2869421126?check_suite_focus=true is fixed with it. I don't know exactly why this error is happening.

christian-bromann commented:
Can we create this log file as part of the test? Have you tried to remove this file on your machine and rerun the tests?

yon-cuadrado commented:
I have executed the tests in my machine without the log file and the tests pass. I have modified the tests to create automatically the logfile in case it doesn't exist and I have added some type fixes that where making the package fail when it was used from another project

wswebcreation commented on package.json:
```diff
@@ -2,17 +2,22 @@
   "name": "wdio-cucumberjs-json-reporter",
   "version": "2.0.4",
```
Hi @yon-cuadrado 

Thanks for the PR and all the work you've put into it. I just want to check-check-double-check before we merge this. The current version is version 3.0.0, and the package.json is at least at `2.0.4`, which one was your main branch? Are you sure all latest changed are in this one?
If so, can you upgrade it to 3.0.0 so the package.json file is also in line with the main branch?

Thans

wswebcreation commented on package.json:
```diff
@@ -2,17 +2,22 @@
   "name": "wdio-cucumberjs-json-reporter",
   "version": "2.0.4",
```
Hi @yon-cuadrado 

Thanks for the PR and all the work you've put into it. I just want to check-check-double-check before we merge this. The current version is version 3.0.0, and the package.json is at least at `2.0.4`, which one was your main branch? Are you sure all latest changed are in this one?
If so, can you upgrade it to 3.0.0 so the package.json file is also in line with the main branch?

Thanks

yon-cuadrado replied:
I used the branch main as the starting point for the pr. I have left the version value in the package.json as it is in this branch, what value should I put?, 3.0.1 ?

yon-cuadrado replied:
Hello
Thanks for taking the time to review the pr.
The branch from where I started my pr is main. I have updated the package.json version to 3.0.0. I think that I have all the changes included but as this is my first pr in the project I think it would be great if you could take a closer look just in case I have missed any other thing.

Regards

wswebcreation commented:
@yon-cuadrado 

It really looks great, thanks for all the work. I'll do a last check tomorrow morning and will merge it afterwards.

wswebcreation reviewed (approved):
Thank you very much!!
